### DISCUSSION TITLE
Filesystem API hooks

### DISCUSSION THREAD

**Discussion by aonghas** [pocketbase/pocketbase#4781] [Q&A]
Hi there,

Is there a way I can create an API hook to listen for new files that are added to a subfolder in pb_public? I have a process that can only upload files via FTP so I either want an event hook to create a PB record for each file that's uploaded (could just be a filename) or otherwise some kind of API route to display all the files in a folder directory. I'm looking into the [filesystem](https://pocketbase.io/jsvm/modules/filesystem.html) JVSM docs but any help would be appreciated!

Thanks!
Aonghas

**Comment by ganigeorgiev** [comment]
No, at the moment there are no file hooks but this will be supported as part of the refactoring and https://github.com/pocketbase/pocketbase/issues/2387.

**Reply by ganigeorgiev** [reply]
Also note that uploaded files are not stored in `pb_public` but in `pb_data/storage` when the local filesystem is used.

**Reply by ganigeorgiev** [reply]
On second read, the planned hooks may not work for your case.

It seems that you want to watch files that are dynamically added/removed/changed in `pb_public`. You can use [fsnotify](https://github.com/fsnotify/fsnotify) or other similar library but this is not available in the JSVM.

**Reply by aonghas** [reply]
Oh it doesn't have to be pb_public so long as the hook can trigger a record to be created with the file as an attachment so then it can be accessed via API. If this will be possible via the refactoring that would be awesome!
